src/vs/workbench/api/common/extHostOutput.ts

```diff
@@ -20,18 +21,20 @@ export abstract class AbstractExtHostOutputChannel extends Disposable implements
 	protected readonly _proxy: MainThreadOutputServiceShape;
 	private _disposed: boolean;
 	private _offset: number;
+	private _extension: IExtensionDescription | undefined;
 
 	protected readonly _onDidAppend: Emitter<void> = this._register(new Emitter<void>());
 	readonly onDidAppend: Event<void> = this._onDidAppend.event;
 
-	constructor(name: string, log: boolean, file: URI | undefined, extensionId: string | undefined, proxy: MainThreadOutputServiceShape) {
+	constructor(name: string, log: boolean, file: URI | undefined, extension: IExtensionDescription | undefined, proxy: MainThreadOutputServiceShape) {
```

Comment:
I assume the ` | undefined` portion is for testing? For such cases there exists `nullExtensionDescription` which can than enable stricter typings

Reply:
It wasn't - I am not that familiar with the codebase. Will update the type. Thanks for the feedback :)

Reply:
Updated to use `nullExtensionDescription` and changed type to be `IExtensionDescription` instead of `IExtensionDescription | undefined`.